cairo-contracts/packages/apps/src/transfer/interfaces/transfer.cairo

```diff
@@ -25,5 +25,8 @@ pub trait ITransferQuery<TContractState> {
     /// ```
     /// Hashing the denom is delegated to the client as it is more cost-efficient.
     fn ibc_token_address(self: @TContractState, token_key: felt252) -> ContractAddress;
+
+    /// Return the contract addresses of all IBC tokens.
+    fn ibc_token_addresses(self: @TContractState) -> Array<ContractAddress>;
 }
```

Comment:
If the goal of adding this endpoint is only to obtain the token address of an IBC-created token, it’s actually unnecessary. We can either listen to the `CreateTokenEvent` that gets emitted or construct the token's key on the relayer side and use the `ibc_token_address`, same as it's [computed](https://github.com/informalsystems/ibc-starknet/blob/9694808e28d4cb963abe6a68ee30bfa4d2c1d645/cairo-contracts/packages/apps/src/transfer/types.cairo#L120-L130) by the contract.

Reply:
It is indeed unnecessary. I added it as a hack. I just forgot to add a `TODO` comment on top of it.

Essentially, the `relay_packet` method should return any stateful data generated during the relay -- specifically token address of the minted denom. I will wait for Soares to fix this. After that, we can remove this.

Reply:
Could you open up an issue for this? What exactly is blocking us and what's needed in `hermes-sdk` so that we can rely on existing facilities here?

Reply:
#175